pulp_smash/tests/pulp2/rpm/api_v2/test_upload_publish.py

```diff
@@ -468,3 +468,56 @@ def _verify_search_results(self, units):
             units[0]['metadata']['vendor'],
             RPM_WITH_VENDOR_DATA['metadata']['vendor'],
         )
+
+
+class UploadInvalidRPMTestCase(unittest.TestCase):
+    """Test whether one invalid RPM upload fails and produce error details."""
+
+    def test_all(self):
+        """Test whether one invalid RPM upload fails and produces error details.
+
+        This test targets the following issues.
+
+        * `Pulp Smash #544 <https://github.com/PulpQE/pulp-smash/issues/544>`_
+        * `Pulp issue #2543 <https://pulp.plan.io/issues/2543>`_
+        * `Pulp issue #3090 <https://pulp.plan.io/issues/3090>`_
+
+        Do the following:
+
+        1. Create a RPM repository.
+        2. Create an invalid RPM package.
+        3. Upload an invalid RPM to repository.
+        4. Assert that error message is descriptive of the action that
+           triggered the error.
+        5. Assert that upload fails.
+        """
+        cfg = config.get_config()
+        if selectors.bug_is_untestable(2543, cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/2543')
+        if selectors.bug_is_untestable(3090, cfg.pulp_version):
+            self.skipTest('https://pulp.plan.io/issues/3090')
+        client = api.Client(cfg, api.json_handler)
+        repo = client.post(REPOSITORY_PATH, gen_repo())
+        self.addCleanup(client.delete, repo['_href'])
+
+        # Create invalid RPM
+        cli_client = cli.Client(cfg, cli.code_handler)
+        sudo = '' if utils.is_root(cfg) else 'sudo '
```

Comment:
Why should `touch ....rpm` be executed as root? Unnecessary privilege escalation is unnecessary.